go/private/actions/compile.bzl

```diff
@@ -65,3 +65,28 @@ def emit_compile(ctx, go_toolchain,
       arguments = args,
       env = go_toolchain.env,
   )
+
+
+def bootstrap_compile(ctx, go_toolchain,
+    sources = None,
+    importpath = "",
+    golibs = [],
+    mode = NORMAL_MODE,
+    out_lib = None,
+    gc_goopts = []):
+  """See go/toolchains.rst#compile for full documentation."""
+
+  if sources == None: fail("sources is a required parameter")
+  if out_lib == None: fail("out_lib is a required parameter")
+  if golibs:  fail("compile does not accept deps in bootstrap mode")
```

Comment:
Check for non-default values in all unused parameters. Let's be careful not to silently ignore things that shouldn't be specified.

Reply:
The only thing not used is mode, which we need to accept but ignore for it to work.